src/steps/02-EvaluationCriteria/MRR/ReviewBarriers.vue

```diff
@@ -7,19 +7,187 @@
+    ATATSVGIcon
+  },
+})
 
-export default class ReviewBarriers extends Vue {}
+export default class ReviewBarriers extends Mixins(SaveOnLeave){
+  /* eslint-disable camelcase */
+  public barriersToOpportunity =""
+  public agency = ""
+  public insertDate = ""
+  public previousJA = ""
+  public defaultSuggestion = ""
+  public showRestoreModal = false;
+  public writeOwnExplanation = false;
+  public isBarriersToOpportunityIsDefault = false;
+
+
+  public get getRowCount(): number {
+    return this.writeOwnExplanation ? 12 : 19;
+  }
+
+  @Watch("barriersToOpportunity")
+  public barriersToOpportunityChanged(): void {
+    this.isBarriersToOpportunityIsDefault = this.barriersToOpportunity === this.defaultSuggestion;
+  }
+  public restoreSuggestion(): void {
+    this.barriersToOpportunity = this.defaultSuggestion;
+    this.showRestoreModal = false;
+  }
+
+  public confirmRestoreDefaultText(): void {
+    this.showRestoreModal = true;
+  }
+  get btnRestoreIconColor(): string {
+    return this.isBarriersToOpportunityIsDefault ? "disabled" : "primary";
+  }
+
+  private get currentData(): FairOpportunityDTO {
+    return {
+      barriers_plans_to_remove_custom: this.barriersToOpportunity,
+      barriers_plans_to_remove_generated: this.defaultSuggestion
+    } as FairOpportunityDTO;
+  }
+
+  private get savedData(): FairOpportunityDTO {
+    return {
+      barriers_plans_to_remove_custom: AcquisitionPackage.fairOpportunity?.
+        barriers_plans_to_remove_custom,
+      barriers_plans_to_remove_generated: AcquisitionPackage.fairOpportunity?.
+        barriers_plans_to_remove_generated,
+    } as FairOpportunityDTO;
+  }
+
+  private hasChanged(): boolean {
+    return hasChanges(this.currentData, this.savedData);
+  }
+
+  public async loadOnEnter(): Promise<void> {
+    const agencyData = convertAgencyRecordToSelect(OrganizationData.agency_data);
+    const organizationStore = await AcquisitionPackage
+      .loadData<OrganizationDTO>({storeProperty:
+        StoreProperties.Organization}) as Record<string, any>;
+    const selectedAgencyIndex = agencyData.findIndex(
+      (svc) => svc.value === organizationStore.agency.value
+    );
+    if (selectedAgencyIndex > -1) {
+      this.agency = agencyData[selectedAgencyIndex].text || "";
+    }
+    const storeData = _.cloneDeep(AcquisitionPackage.fairOpportunity);
+    if(storeData){
+      this.insertDate = storeData.barriers_follow_on_expected_date_awarded||""
+      this.previousJA = storeData.barriers_j_a_prepared_results||""
+
+      if(storeData.barriers_follow_on_requirement === "YES"){
+        this.defaultSuggestion += "To overcome future barriers to competition," +
+           this.agency + " is preparing a fair opportunity competitive" +
+          " follow-on requirement. The follow-on is expected to be" +
+          " completed, solicited, and awarded by " + this.insertDate + "."
+      }
+      if(storeData.barriers_agency_pursuing_training_or_certs === "YES"){
```

Comment:
Inserting this comment here but this applies to all the 3 if conditions you have for compiling "defaultSuggestion". Looking at the story, it looks like the 3 paragraphs of the defaultSuggestion should always be included. Please double check this. Depending on how this goes, you PR may change significantly. I will continue my review once you resolve the 2 comments that I added.

Reply:
the Figma shows that these are added in depending on the values captured in the previous screen.